### DISCUSSION TITLE
How should i modify the .desktop file to make it compatible with st (suckless terminal)

### DISCUSSION THREAD

**Discussion by Kenneth-Mata** [helix-editor/helix#4951] [Q&A] [answered]
Hi, i get Helix from the COPR repo and i'm trying to get it to work with xdg, however following the instructions in the readme (which, tbf are from kitty) doesn't seem to work, i've tried ``st hx %F``, ``st -e hx %F``, ``st -e hx`` and ``st hx`` in the ``Exec`` to no avail, (i also used the ``Terminal=false`` option)

**Comment by David-Else** [comment, marked as answer]
I wrote the original info in the README:

- Are you using Gnome?
 
I have tested in Gnome and it works fine for kitty. I was told it should work in other desktops, but have not personally tested it.

All the `sed` command does is replace `hx` with `kitty hx` and set using the default terminal to false (you need to do that). Replacing `kitty hx` with `[name of your terminal executable in the path] hx` should do it.

- Have you tried editing `Helix.desktop` manually?
 
The problem might be after you have run `sed` and changed the file then it is not going to be the same and the find replace might not work. 

My current `Helix.desktop`  is:
```
Exec=kitty hx %F
Terminal=false
```
I assume you want:
```
Exec=st hx %F
Terminal=false
```
- Are you sure `st` is in the $PATH, you can call it from another terminal?

**Comment by Kenneth-Mata** [comment]
Thank you, that solves it, I use a keybinding on my wm to call st with helix ``st -e hx`` so that's why i tried with that.
